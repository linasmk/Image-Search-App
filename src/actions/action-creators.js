/* ========= Dependencies ============= */
import { v4 as uuidv4 } from "uuid";
import axios from "axios";
/* ========= Imports ============= */
import { UnsplashAccessKey } from "@root/accessKey";
import Unsplash, { toJson } from "unsplash-js"; // eslint-disable-line no-unused-vars
import {
  ADD_SAVED_QUERY,
  REMOVE_SAVED_QUERY,
  GET_IMAGES
} from "@consts/constants";
/* =================================================
=================== ACTION CREATORS ================
================================================= */
export const getImages = (img) => ({
  type: GET_IMAGES,
  img
});

export const addSavedQuery = ({ name = "" } = {}) => ({
  type: ADD_SAVED_QUERY,
  savedQuery: {
    id: uuidv4(),
    name
  }
});

export const removeSavedQuery = ({ id } = {}) => ({
  type: REMOVE_SAVED_QUERY,
  id
});
