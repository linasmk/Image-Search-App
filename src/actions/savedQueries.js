/* ========= Dependencies ============= */
import { v4 as uuidv4 } from "uuid";
/* ========= Imports ============= */
import { ADD_SAVED_QUERY, REMOVE_SAVED_QUERY } from "@consts/constants";
/* =================================================
=========== ACTION CREATORS: SAVED QUERIES ========
================================================= */

export const addSavedQuery = ({ name = "" } = {}) => ({
  type: ADD_SAVED_QUERY,
  savedQuery: {
    id: uuidv4(),
    name
  }
});

export const removeSavedQuery = ({ id } = {}) => ({
  type: REMOVE_SAVED_QUERY,
  id
});
