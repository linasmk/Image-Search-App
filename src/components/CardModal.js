/* ========= App Dependencies ============= */
import React from "react";
import PropTypes from "prop-types";
import { VscChromeClose } from "react-icons/vsc";
import Modal from "react-modal";
/* ========= Code ============= */

export const CardModal = ({
  openCardModal,
  closeCardModal,
  altDescription,
  imageID,
  likes,
  urlsFull,
  firstName,
  lastName,
  download,
}) => {
  const capitalizeFirstLetter = (string) =>
    string.charAt(0).toUpperCase() + string.slice(1);

  return (
    <Modal
      isOpen={openCardModal}
      onRequestClose={closeCardModal}
      contentLabel={altDescription}
      closeTimeoutMS={800}
      className="card-modal"
      key={imageID}
      style={{
        overlay: {
          backgroundColor: "rgba(0, 0, 0, .75)",
        },
        content: {
          color: "lightsteelblue",
        },
      }}
      ariaHideApp={false}>
      <div className="card-modal__content--wrapper">
        <VscChromeClose
          className="card-modal__close-btn"
          onClick={closeCardModal}
        />
        <article className="card-modal__content">
          <div className="card-modal__topbar" />
          <section className="card-modal__description-wrapper">
            <h2 className="card-modal__description">
              {altDescription != null
                ? capitalizeFirstLetter(altDescription)
                : ""}
            </h2>
          </section>
          <section className="card-modal__main">
            <img
              className="card-modal__img"
              alt={altDescription}
              src={urlsFull}
            />
          </section>
          <section className="card-modal__info-bar">
            <p className="card-modal__data">
              {likes != null ? `${likes} ` : 0}
              <span className="card-modal__text">Likes</span>
            </p>
            <p className="card-modal__data">
              <span className="card-modal__text">by </span>
              {firstName !== null ? firstName : `Uknown`}
              <span> </span>
              {lastName !== null ? lastName : `Uknown`}
            </p>
          </section>
          <section className="card-modal__btn-bar">
            <a
              className="card-modal__btn--download"
              href={`${download}?force=true`}
              download
              rel="noreferrer"
              target="_blank">
              Download
            </a>
          </section>
        </article>
      </div>
    </Modal>
  );
};
CardModal.defaultProps = {
  firstName: null,
  lastName: null,
  altDescription: null,
  likes: 0,
};
CardModal.propTypes = {
  openCardModal: PropTypes.bool.isRequired,
  closeCardModal: PropTypes.func.isRequired,
  imageID: PropTypes.string.isRequired,
  urlsFull: PropTypes.string.isRequired,
  download: PropTypes.string.isRequired,
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  altDescription: PropTypes.string,
  likes: PropTypes.number,
};
export default CardModal;
