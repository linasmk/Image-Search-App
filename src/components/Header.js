/* ===== App Dependencies ===== */
import React from "react";
/* ============ Code ============ */

const Header = () => (
  <header className="header">
    <div className="header__inner-wrapper">
      <h1 className="title">Image Search App</h1>
    </div>
  </header>
);
export default Header;
