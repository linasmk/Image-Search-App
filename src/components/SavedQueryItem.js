/* ========= App Dependencies ============= */
import React from "react";
import PropTypes from "prop-types";

/* ========= Code ============= */
export const SavedQueryItem = ({ removeSavedQuery, onClick, name, id }) => {
  /* ========== Functions ========== */
  const removeQuery = () => removeSavedQuery({ id });

  const handleChildQuery = (e) => {
    onClick(e.target.textContent);
  };

  /* ========== Content ========== */
  return (
    <div className="saved-queries__item">
      <p
        className="saved-queries__txt"
        role="presentation"
        onClick={handleChildQuery}>
        {name}
      </p>
      <button
        type="button"
        className="saved-queries__btn--remove"
        onClick={removeQuery}>
        &#88;
      </button>
    </div>
  );
};
SavedQueryItem.defaultProps = {
  onClick: undefined,
  removeSavedQuery: undefined
};
SavedQueryItem.propTypes = {
  removeSavedQuery: PropTypes.func,
  onClick: PropTypes.func,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};
export default SavedQueryItem;
