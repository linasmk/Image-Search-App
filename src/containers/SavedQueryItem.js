/* ===== Redux ===== */
import { connect } from "react-redux";
import { removeSavedQuery } from "../actions/savedQueries";
/* ====== Components ===== */
import SavedQueryItemComponent from "../components/SavedQueryItem";
/* ========= Code ========= */
const mapDispatchToProps = (dispatch) => ({
  removeSavedQuery: (id) => dispatch(removeSavedQuery(id)),
});

export const SavedQueryItem = connect(
  (state, props) => ({
    savedQueries: state.savedQueries.find(
      (savedQuery) => savedQuery.id === props.id,
    ),
  }),
  mapDispatchToProps,
)(SavedQueryItemComponent);
