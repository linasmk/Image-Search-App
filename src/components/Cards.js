/* ========= App Dependencies ============= */
import React, { useState } from "react";
import { cl } from "../utils";
/* ========= Components =============== */
import CardModal from "./CardModal";
import Loader from "./Loader";
/* ========= Code ============= */

export const Cards = (props) => {
  const [cardModalId, setCardModalId] = useState(undefined);
  const openCardModal = (id) => setCardModalId(id);
  const closeCardModal = () => setCardModalId(undefined);

  return (
    <div>
      {props.topLoader ? (
        <Loader display="block" />
      ) : (
        <section className="card-list">
          {props.noResultsError ? (
            <section className="no-results">
              <h3 className="no-results__warning">
                Your search keyword did not return any results!
              </h3>
            </section>
          ) : (
            props.images.map((image, index) => (
              <div className="card" key={index}>
                <img
                  className="card--image"
                  onClick={() => openCardModal(image.id)}
                  alt={image.alt_description}
                  src={image.urls.small}
                  width="50%"
                  height="50%"></img>
                <CardModal
                  openCardModal={cardModalId === image.id}
                  closeCardModal={closeCardModal}
                  {...image}
                />
              </div>
            ))
          )}
        </section>
      )}
    </div>
  );
};
export default Cards;
