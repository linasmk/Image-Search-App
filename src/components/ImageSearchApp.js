/* ========= App Dependencies ============= */
import React from "react";
/* ========= Components ============= */
import Header from "@comp/Header";
import { SearchImages } from "@cont/SearchImages";
import Footer from "@comp/Footer";
/* ========= Code ============= */
const ImageSearchApp = () => (
  <div className="image-search-app">
    <Header />
    <SearchImages />
    <Footer />
  </div>
);
export default ImageSearchApp;
