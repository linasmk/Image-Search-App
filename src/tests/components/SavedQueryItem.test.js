/* ========= App Dependencies ============= */
import React from "react";
import { shallow } from "enzyme";
import { removeSavedQuery } from "../../actions/savedQueries";
import { SavedQueryItem } from "../../components/SavedQueryItem";
import savedQueries from "../fixtures/savedQueries";
/* ========= Code ============= */

it("renders SavedQueryItem correctly", () => {
  const removeSavedQuery = jest.fn();
  const wrapper = shallow(
    <SavedQueryItem {...savedQueries[0]} removeSavedQuery={removeSavedQuery} />
  );
  expect(wrapper).toMatchSnapshot();
});
