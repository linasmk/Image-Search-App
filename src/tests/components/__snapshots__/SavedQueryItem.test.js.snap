// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SavedQueryItem correctly 1`] = `
<div
  className="saved-queries__item"
>
  <p
    className="saved-queries__txt"
    onClick={[Function]}
  >
    dog
  </p>
  <button
    className="saved-queries__btn"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
